LearningMissionSimulation/PlaygroundMher.cs

```diff
@@ -219,5 +245,31 @@ public void RegisterStudentsForClasses()
             throw new NotImplementedException();
         }
 
+        #region Reports
+        void ReportHeader(string actionName)
+        {
+            Console.WriteLine($"******{actionName} is started******\n");
+        }
+
+        void ReportFooter(string actionName)
+        {
+            Console.WriteLine($"******{actionName} is finished******\n");
+        }
+
+        void ReportItem(string itemName, string actionName, int itemIndex)
+        {
+            Console.WriteLine($" {actionName} {itemIndex}\n{itemName}");
+        }
+
+        void ReportSummary(string summary, int count)
+        {
+            Console.WriteLine($"''''''Generated {count} {summary}''''''\n");
```

Comment:
Rename the "summary" parameter to "actionName". Make the method more generic by removing word "Generated"
It should be something like:
void ReportSummary(string itemName, string actionName, int itemCount)
{
            Console.WriteLine($"'''''' {actionName}  {itemCount}  {itemName}''''''\n");
}